stag-library-compiler/src/test/java/com/vimeo/stag/processor/functional/JavaIntegrationFunctionalTests.kt

```diff
@@ -150,8 +151,49 @@ class JavaIntegrationFunctionalTests {
         assertThatClassCompilationIsSuccessful(WrapperTypeAdapterModel::class)
     }
 
+    @Test
+    fun `Verify that compilation is deterministic`() {
+        val classes = arrayOf(
+                AlternateNameModel::class,
+                AlternateNameModel1::class,
+                BaseExternalModel::class,
+                BooleanFields::class,
+                EnumWithFieldsModel::class,
+                ExternalAbstractClass::class,
+                ExternalModel1::class,
+                ExternalModel2::class,
+                ExternalModelGeneric::class,
+                ExternalModelGeneric1::class,
+                ModelWithNestedInterface::class,
+                NativeJavaModel::class,
+                NativeJavaModelExtension::class,
+                NativeJavaModelExtensionWithoutAnnotation::class,
+                RawGenericField::class,
+                NullFields::class,
+                PrivateMembers::class,
+                WildcardModel::class,
+                DynamicallyTypedModel::class,
+                DynamicallyTypedWildcard::class,
+                AbstractDataList::class,
+                SuperAbstractDataList::class,
+                ConcreteDataList::class
+        )
+
+        val compilation1Hash = processorTester.compileClassesInModule(module, *classes).hashOutput()
+        val compilation2Hash = processorTester.compileClassesInModule(module, *classes).hashOutput()
+
+        assertThat(compilation1Hash).isEqualTo(compilation2Hash)
+    }
+
+    /**
+     * Returns the concatenation of the hash of each file generated by the [Compilation].
+     */
+    private fun Compilation.hashOutput(): String = generatedFiles()
+            .map { it.getCharContent(false).hashCode() }
+            .joinToString()
```

Comment:
Optional: `joinToString` can take a `transform`, so you could move the logic from `map` and just use the `joinToString`... 

```kotlin
.joinToString { it.getCharContent(false).hashCode().toString() }`
```